frontend/src/lib/utils.tsx

```diff
@@ -967,6 +968,54 @@ export function dateStringToDayJs(date: string | null): dayjs.Dayjs | null {
     return response
 }
 
+export const getDefaultInterval = (dateFrom: string | null, dateTo: string | null): IntervalType => {
+    // use the default mapping if we can
+    for (const mapping of dateMapping) {
+        const mappingFrom = mapping.values[0] ?? null
+        const mappingTo = mapping.values[1] ?? null
+        if (mappingFrom === dateFrom && mappingTo === dateTo && mapping.defaultInterval) {
+            return mapping.defaultInterval
+        }
+    }
+
+    if (dateFrom?.endsWith('h') || dateTo?.endsWith('h')) {
+        return 'hour'
+    }
+
+    if (
+        dateFrom === 'mStart' ||
+        dateFrom?.endsWith('d') ||
+        dateFrom?.endsWith('dStart') ||
+        dateFrom?.endsWith('dEnd') ||
+        dateTo?.endsWith('d') ||
+        dateTo?.endsWith('dStart') ||
+        dateTo?.endsWith('dEnd')
+    ) {
+        return 'day'
+    }
+
+    if (
+        dateFrom === 'all' ||
+        dateFrom === 'yStart' ||
+        dateFrom?.endsWith('m') ||
+        dateFrom?.endsWith('mStart') ||
+        dateFrom?.endsWith('mEnd') ||
+        dateFrom?.endsWith('y') ||
+        dateFrom?.endsWith('yStart') ||
+        dateFrom?.endsWith('yEnd') ||
+        dateTo?.endsWith('m') ||
+        dateTo?.endsWith('mStart') ||
+        dateTo?.endsWith('mEnd') ||
+        dateTo?.endsWith('y') ||
+        dateTo?.endsWith('yStart') ||
+        dateTo?.endsWith('yEnd')
+    ) {
```

Comment:
I wonder if it would be conceptually simpler to extract these lines from `dateStringToDayJs`:

```ts
    const parseDate = /^([-+]?)([0-9]*)([dmwqy])(|Start|End)$/
    const matches = (date || '').match(parseDate)
    const [, sign, rawAmount, rawUnit, clip] = matches
    const amount = rawAmount ? parseInt(sign + rawAmount) : 0
    const unit = dateOptionsMap[rawUnit] || 'day'
```

... into an `extractDateString` method, and use that to just compare the units? This does look a bit messy. 

Also, I'd introduce quarters here. This function is a common utility function, so you'll never know where it'll be used. Might as well make it feature complete with all the variants of `dateOptionsMap`.

Duplicated logic in almost identical date functions that slowly creeps out of date is something we have experience with here... but so far only on the python side. 😅

Reply:
I did this :)